WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java

```diff
@@ -959,4 +959,12 @@ public static void setSystemNotificationsEnabled(boolean enabled) {
     public static boolean getSystemNotificationsEnabled() {
         return getBoolean(UndeletablePrefKey.SYSTEM_NOTIFICATIONS_ENABLED, true);
     }
+
+    public static boolean getMasterSwitchKeyEnabled(@NonNull String masterKey) {
+        return prefs().getBoolean(masterKey, true);
```

Comment:
I'm wondering why we need to set this in prefs. Isn't it enough to base the state of the master switch on the child preferences? I might be missing something 😄

Reply:
I was thinking if it is a shared preference key we could probably do that in the custom view via `AppPrefs` which stores keys in shared preferences instantiated using application context and can be accessed anywhere in the application e.g. a model class like `Note` which might not have a `context`.

Reply:
Generally we have a way of storing preferences and this one is different (which might cause confusion). The switch (especially the one in the dialog) will have different value for different sites so the previously kept state is not useful. In this case you'll have value per followed site. In other cases it might be per your site or something else. I don't think this value should be stored in the preferences by the View key. 
It might be only useful if the preference is unique per user (for the deletable preferences) or per installation (for non-deletable preferences). In this case it might lead to bugs.